lib/menu.rb

```diff
@@ -0,0 +1,78 @@
+require_relative 'order'
+require_relative 'delivery_sms'
+
+class Menu
+
+  MENU = [
+    { name: "Bertie Bott's Every Flavour Beans", price: 7 },
+    { name: "Cauldron Cakes", price: 6 },
+    { name: "Chocolate Frogs", price: 5 },
+    { name: "Drooble’s Best Blowing Gum", price: 10 },
+    { name: "Pumpkin Pasties", price: 5 },
+    { name: "Licorice Wands", price: 9 }
+  ]
+
+  def initialize(order = Order.new, sms = DeliverySMS.new)
+    @order = order
+    @sms = sms
+  end
+
+  def see
+    print_out
+  end
+
+  def choose(dish, quantity)
+    dish_price_quantity = search_menu(dish, quantity)
+    order.add(dish_price_quantity)
+  end
+
+  def confirm_order(customer_input_price)
+    order.confirm(customer_input_price)
+  end
+
+  def see_sms
+    sms.send_by_sms(print_out_sms)
+  end
+
+  private
+
+  attr_reader :menu, :order, :sms
+
+  def search_menu(dish, quantity)
+    dish_price_quantity = []
+    MENU.each do |item|
+      dish_price_quantity = item if dish == item[:name]
+    end
+    dish_price_quantity[:quantity] = quantity
+    dish_price_quantity
+  end
+
+  def print_out
```

Comment:
Should formatting the menu be a responsibility of Menu or could this be split out into a different class? Think there is some duplication between Menu and Order that could be split into formatter classes and a printer class.

Reply:
In general it feels like Menu has several responsibilities - could this be simplified or delegated?